src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -471,10 +523,13 @@ bool ClassListParser::is_matching_cp_entry(constantPoolHandle &pool, int cp_inde
   }
   return true;
 }
-void ClassListParser::resolve_indy(Symbol* class_name_symbol, TRAPS) {
+
+void ClassListParser::resolve_indy(Thread* current, Symbol* class_name_symbol) {
```

Comment:
Should current be a JavaThread here?

Reply:
Currently all the `current` parameters will come from the THREAD/TRAPS of its caller, like

```
int ClassListParser::parse(TRAPS) {
    ...
    resolve_indy(THREAD, class_name_symbol);
```

I could change the caller to use THREAD->as_JavaThread(), but I think that's too much ceremony. It's better to change to `JavaThread*` after THREAD/TRAPS are changed to `JavaThread` (JDK-8252685)